esphome/components/tuya/tuya.h

```diff
@@ -34,14 +34,24 @@ struct TuyaDatapointListener {
 
 enum class TuyaCommandType : uint8_t {
   HEARTBEAT = 0x00,
-  QUERY_PRODUCT = 0x01,
-  MCU_CONF = 0x02,
+  PRODUCT_QUERY = 0x01,
+  CONF_QUERY = 0x02,
   WIFI_STATE = 0x03,
   WIFI_RESET = 0x04,
   WIFI_SELECT = 0x05,
-  SET_DATAPOINT = 0x06,
-  STATE = 0x07,
-  QUERY_STATE = 0x08,
+  DP_DELIVER = 0x06,
```

Comment:
Hmm, I don't really like the `dp` abbreviation. At least when I read the code initially for this component, it took me a while to understand that dp stood for datapoint. And besides its written out fully everywhere else here too.

```suggestion
  DATAPOINT_DELIVER = 0x06,
```

Reply:
If you're going to duplicate the values like that, might as well separate them into send and receive groups, so it's more clear.  In most cases, the same value is used for the request and the response.

Reply:
Tuya uses the DP initialism all over the place (dpid, etc) so I kinda got used to it. I can expand it out. I think keeping them in order of value and not duplicating for rx and tx is fine, since (as you noted) the same command code is valid for both sides of the transaction.